src/main/webapp/app/entities/conflict/conflict.tsx

```diff
@@ -1,36 +1,134 @@
 import React from 'react';
 import { connect } from 'react-redux';
 import { Link, RouteComponentProps } from 'react-router-dom';
-import { Button, Col, Row, Table } from 'reactstrap';
+import { Button, Col, Row, Table, Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
 // tslint:disable-next-line:no-unused-variable
-import { Translate, ICrudGetAllAction, TextFormat } from 'react-jhipster';
+import {
+  Translate,
+  translate,
+  ICrudGetAllAction,
+  TextFormat,
+  JhiPagination,
+  getPaginationItemsNumber,
+  getSortState,
+  IPaginationBaseState
+} from 'react-jhipster';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
 import { IRootState } from 'app/shared/reducers';
-import { getEntities } from './conflict.reducer';
+import { getEntities, updateEntity } from './conflict.reducer';
+import { ITEMS_PER_PAGE_ENTITY } from 'app/shared/util/pagination.constants';
 import { IConflict } from 'app/shared/model/conflict.model';
 // tslint:disable-next-line:no-unused-variable
 import { APP_DATE_FORMAT, APP_LOCAL_DATE_FORMAT } from 'app/config/constants';
 
 export interface IConflictProps extends StateProps, DispatchProps, RouteComponentProps<{ url: string }> {}
 
-export class Conflict extends React.Component<IConflictProps> {
+export interface IConflictState extends IPaginationBaseState {
+  dropdownOpenTop: boolean;
+  dropdownOpenBottom: boolean;
+  itemsPerPage: number;
+}
+
+export class Conflict extends React.Component<IConflictProps, IConflictState> {
+  constructor(props) {
+    super(props);
+
+    this.toggleTop = this.toggleTop.bind(this);
+    this.toggleBottom = this.toggleBottom.bind(this);
+    this.select = this.select.bind(this);
+    this.state = {
+      dropdownOpenTop: false,
+      dropdownOpenBottom: false,
+      itemsPerPage: ITEMS_PER_PAGE_ENTITY,
+      ...getSortState(this.props.location, ITEMS_PER_PAGE_ENTITY)
+    };
+  }
+
   componentDidMount() {
-    this.props.getEntities();
+    this.getEntities();
+  }
+
+  sort = prop => () => {
+    this.setState(
+      {
+        order: this.state.order === 'asc' ? 'desc' : 'asc',
+        sort: prop
+      },
+      () => this.sortEntities()
+    );
+  };
+
+  sortEntities() {
+    this.getEntities();
+    this.props.history.push(`${this.props.location.pathname}?page=${this.state.activePage}&sort=${this.state.sort},${this.state.order}`);
+    window.scrollTo(0, 0);
+  }
+
+  handlePagination = activePage => this.setState({ activePage }, () => this.sortEntities());
+
+  getEntities = () => {
+    const { activePage, itemsPerPage, sort, order } = this.state;
+    this.props.getEntities(activePage - 1, itemsPerPage, `${sort},${order}`);
+  };
+
+  toggleTop() {
+    this.setState(prevState => ({
+      dropdownOpenTop: !prevState.dropdownOpenTop
+    }));
+  }
+
+  toggleBottom() {
+    this.setState(prevState => ({
+      dropdownOpenBottom: !prevState.dropdownOpenBottom
+    }));
+  }
+
+  select(event, activePage) {
```

Comment:
I think it will look better if you will have array with values [10,20,50,100] and for each you will create DropdownItem with this.select(value) on onClick instead of event.target.innerText.